### DISCUSSION TITLE
How to set 404 statusCode for current page?

### DISCUSSION THREAD

**Discussion by Innei** [nuxt/nuxt#16302] [Questions] [answered]
Hi guys!

If some data not found, I want to stay on the current page (layout) but with 404 statusCode, without redirection. Is it possible?

In nuxt2, I can do this.

```
async asyncData({ res}) {
    const { data } = await axios.get('...')
    if (!data) {
        res.statusCode = 404
    }
}
```

But if I use `useAsyncData` hook, how can I rewrite response status.

**Comment by TheAlexLichter** [comment, marked as answer]
You can use the  [`setResponseStatus` utility](https://nuxt.com/docs/api/utils/set-response-status#setresponsestatus).
